deps/rabbit/src/rabbit_amqp_filtex.erl

```diff
@@ -119,9 +119,11 @@ validate0(Descriptor, KVList0) when
     KVList = lists:map(fun({{utf8, Key}, {utf8, String}}) ->
                                {Key, parse_string_modifier_prefix(String)};
                           ({{utf8, Key}, TaggedVal}) ->
-                               {Key, unwrap(TaggedVal)}
+                               {Key, unwrap(TaggedVal)};
+                          (_) ->
+                               invalid_filter
```

Comment:
Your modification violates the spec when sending the `filter` field from RabbitMQ to the consumer:
> the sending endpoint sets the filter actually in place

In your change, the desired filter as sent from the client to RabbitMQ is echoed back from RabbitMQ to the client while RabbitMQ silently modifies the filter.
This function should return `error` if the filter is invalid such that the entire filter doesn't take effect. 
For example, the Java client compares only the filter names that were desired (i.e. sent from consumer to RabbitMQ) with the filter names actually in place (i.e. sent from RabbitMQ to the consumer).